src/util-inl.h

```diff
@@ -219,6 +219,18 @@ bool StringEqualNoCase(const char* a, const char* b) {
   return false;
 }
 
+bool StringEqualNoCaseN(const char* a, const char* b, int length) {
+  for (int i = 0; i < length; i++) {
+    if (ToLower(a[i]) != ToLower(b[i]))
+      return false;
+    if (a[i] == '\0')
```

Comment:
I think we are guaranteed at this point that if `a[i]` is `null`, then `b[i]` is also `null`. I think you can simply return `true` if `a[i]` is `null`. The second `if` below can simply be dropped.

It would be good to add a test that compares: `"abc\0abc"` and `"abc\0xyz"`. Your function should (already does) return true for this. This matches the semantics of `strncmp`.